### ISSUE TITLE
RuntimeError: Circular dependency detected while autoloading constant AirbrakeApi::V3::NoticeParser

### ISSUE THREAD

**Issue by marcelloceschia** [errbit/errbit#1235]
WHERE:
api/v3/notices#create

app/controllers/api/v3/notices_controller.rb:17

``` 
FULL BACKTRACE:
[GEM_ROOT]/gems/activesupport-4.2.9/lib/active_support/dependencies.rb:492
[GEM_ROOT]/gems/activesupport-4.2.9/lib/active_support/dependencies.rb:184
app/controllers/api/v3/notices_controller.rb:27
app/controllers/api/v3/notices_controller.rb:11
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_controller/metal/implicit_render.rb:4
[GEM_ROOT]/gems/actionpack-4.2.9/lib/abstract_controller/base.rb:198
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_controller/metal/rendering.rb:10
[GEM_ROOT]/gems/actionpack-4.2.9/lib/abstract_controller/callbacks.rb:20
[GEM_ROOT]/gems/activesupport-4.2.9/lib/active_support/callbacks.rb:117
[GEM_ROOT]/gems/activesupport-4.2.9/lib/active_support/callbacks.rb:117
[GEM_ROOT]/gems/activesupport-4.2.9/lib/active_support/callbacks.rb:555
[GEM_ROOT]/gems/activesupport-4.2.9/lib/active_support/callbacks.rb:505
[GEM_ROOT]/gems/activesupport-4.2.9/lib/active_support/callbacks.rb:505
[GEM_ROOT]/gems/activesupport-4.2.9/lib/active_support/callbacks.rb:92
[GEM_ROOT]/gems/activesupport-4.2.9/lib/active_support/callbacks.rb:778
[GEM_ROOT]/gems/activesupport-4.2.9/lib/active_support/callbacks.rb:81
[GEM_ROOT]/gems/actionpack-4.2.9/lib/abstract_controller/callbacks.rb:19
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_controller/metal/rescue.rb:29
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_controller/metal/instrumentation.rb:32
[GEM_ROOT]/gems/activesupport-4.2.9/lib/active_support/notifications.rb:164
[GEM_ROOT]/gems/activesupport-4.2.9/lib/active_support/notifications/instrumenter.rb:20
[GEM_ROOT]/gems/activesupport-4.2.9/lib/active_support/notifications.rb:164
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_controller/metal/instrumentation.rb:30
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_controller/metal/params_wrapper.rb:250
[GEM_ROOT]/gems/actionpack-4.2.9/lib/abstract_controller/base.rb:137
[GEM_ROOT]/gems/actionview-4.2.9/lib/action_view/rendering.rb:30
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_controller/metal.rb:196
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_controller/metal/rack_delegation.rb:13
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_controller/metal.rb:237
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_dispatch/routing/route_set.rb:74
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_dispatch/routing/route_set.rb:74
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_dispatch/routing/route_set.rb:43
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_dispatch/journey/router.rb:43
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_dispatch/journey/router.rb:30
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_dispatch/journey/router.rb:30
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_dispatch/routing/route_set.rb:817
[GEM_ROOT]/gems/omniauth-1.2.2/lib/omniauth/strategy.rb:186
[GEM_ROOT]/gems/omniauth-1.2.2/lib/omniauth/strategy.rb:164
[GEM_ROOT]/gems/omniauth-1.2.2/lib/omniauth/strategy.rb:186
[GEM_ROOT]/gems/omniauth-1.2.2/lib/omniauth/strategy.rb:164
[GEM_ROOT]/gems/warden-1.2.4/lib/warden/manager.rb:35
[GEM_ROOT]/gems/warden-1.2.4/lib/warden/manager.rb:34
[GEM_ROOT]/gems/warden-1.2.4/lib/warden/manager.rb:34
[GEM_ROOT]/gems/rack-1.6.8/lib/rack/etag.rb:24
[GEM_ROOT]/gems/rack-1.6.8/lib/rack/conditionalget.rb:38
[GEM_ROOT]/gems/rack-1.6.8/lib/rack/head.rb:13
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_dispatch/middleware/params_parser.rb:27
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_dispatch/middleware/flash.rb:260
[GEM_ROOT]/gems/rack-1.6.8/lib/rack/session/abstract/id.rb:225
[GEM_ROOT]/gems/rack-1.6.8/lib/rack/session/abstract/id.rb:220
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_dispatch/middleware/cookies.rb:560
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_dispatch/middleware/callbacks.rb:29
[GEM_ROOT]/gems/activesupport-4.2.9/lib/active_support/callbacks.rb:88
[GEM_ROOT]/gems/activesupport-4.2.9/lib/active_support/callbacks.rb:778
[GEM_ROOT]/gems/activesupport-4.2.9/lib/active_support/callbacks.rb:81
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_dispatch/middleware/callbacks.rb:27
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_dispatch/middleware/remote_ip.rb:78
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_dispatch/middleware/debug_exceptions.rb:17
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_dispatch/middleware/show_exceptions.rb:30
[GEM_ROOT]/gems/railties-4.2.9/lib/rails/rack/logger.rb:38
[GEM_ROOT]/gems/railties-4.2.9/lib/rails/rack/logger.rb:22
[GEM_ROOT]/gems/request_store-1.1.0/lib/request_store/middleware.rb:8
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_dispatch/middleware/request_id.rb:21
[GEM_ROOT]/gems/rack-1.6.8/lib/rack/methodoverride.rb:22
[GEM_ROOT]/gems/rack-1.6.8/lib/rack/runtime.rb:18
[GEM_ROOT]/gems/activesupport-4.2.9/lib/active_support/cache/strategy/local_cache_middleware.rb:28
[GEM_ROOT]/gems/actionpack-4.2.9/lib/action_dispatch/middleware/static.rb:120
[GEM_ROOT]/gems/rack-1.6.8/lib/rack/sendfile.rb:113
[GEM_ROOT]/gems/railties-4.2.9/lib/rails/engine.rb:518
[GEM_ROOT]/gems/railties-4.2.9/lib/rails/application.rb:165
[GEM_ROOT]/gems/rack-1.6.8/lib/rack/deflater.rb:35
[GEM_ROOT]/gems/rack-1.6.8/lib/rack/content_length.rb:15
[GEM_ROOT]/gems/puma-3.8.2/lib/puma/configuration.rb:224
[GEM_ROOT]/gems/puma-3.8.2/lib/puma/server.rb:600
[GEM_ROOT]/gems/puma-3.8.2/lib/puma/server.rb:435
[GEM_ROOT]/gems/puma-3.8.2/lib/puma/server.rb:299
[GEM_ROOT]/gems/puma-3.8.2/lib/puma/thread_pool.rb:120
[GEM_ROOT]/gems/puma-3.8.2/lib/puma/thread_pool.rb:120
```

**Comment by stevecrozz** [comment]
Never seen that one before. Is there more to the stack trace?

**Comment by marcelloceschia** [comment]
@stevecrozz sorry, I forget to include the backtrace. I've updated the initial comment.
Just as notice, I've updated the application to the latest master version this morning

**Comment by stevecrozz** [comment]
This is very strange. Can you tell me a bit more about how you get here? This happens whenever you send a notice? I'm not able to reproduce this yet, so if you can help walk me through the steps to reproduce this issue it would be great. Also, what version of Ruby do you have?

**Comment by marcelloceschia** [comment]
I'm not sure when this occurred. It looks like that occurred during a notice creation, but I will check this again.
I've manually triggered an notice, but in this case it did not triggered this error. May be it will only occur after a restart. I will watch for this "issue" and I hold you up to date.

ruby version: `2.1.9p490`

Environment:
```
CONTENT_LENGTH	1108
CONTENT_TYPE	text/plain;charset=UTF-8
GATEWAY_INTERFACE	CGI/1.2
HTTP_ACCEPT	*/*
HTTP_ACCEPT_ENCODING	br, gzip, deflate
HTTP_ACCEPT_LANGUAGE	en-gb
HTTP_CONNECTION	close
HTTP_DNT	1
HTTP_HOST	errbit.example.org
HTTP_ORIGIN	https://www.example.org
HTTP_REFERER	https://www.example.org/app/clients
HTTP_USER_AGENT	Mozilla/5.0 (iPhone; CPU iPhone OS 11_0_3 like Mac OS X) AppleWebKit/604.1.38 (KHTML, like Gecko) Version/11.0 Mobile/15A432 Safari/604.1
HTTP_VERSION	HTTP/1.0
HTTP_X_FORWARDED_FOR	8.8.8.8
ORIGINAL_FULLPATH	/api/v3/projects/2/create-notice?key=***key***
ORIGINAL_SCRIPT_NAME	
PATH_INFO	/api/v3/projects/2/create-notice
QUERY_STRING	key=***key***
RAW_POST_DATA	{}
REMOTE_ADDR	192.168.122.1
REQUEST_METHOD	POST
REQUEST_PATH	/api/v3/projects/2/create-notice
REQUEST_URI	/api/v3/projects/2/create-notice?key=***key***
ROUTES_47346047345180_SCRIPT_NAME	
SCRIPT_NAME	
SERVER_NAME	errbit.example.org
SERVER_PORT	80
SERVER_PROTOCOL	HTTP/1.1
SERVER_SOFTWARE	puma 3.8.2 Sassy Salamander
ACTION_CONTROLLER_INSTANCE	#<Api::V3::NoticesController:0x00561f3864ce60>
ACTION_DISPATCH_BACKTRACE_CLEANER	#<Rails::BacktraceCleaner:0x007fabb0023d08>
ACTION_DISPATCH_COOKIES	#<ActionDispatch::Cookies::CookieJar:0x00561f38691470>
ACTION_DISPATCH_COOKIES_DIGEST	
ACTION_DISPATCH_COOKIES_SERIALIZER	hybrid
ACTION_DISPATCH_ENCRYPTED_COOKIE_SALT	encrypted cookie
ACTION_DISPATCH_ENCRYPTED_SIGNED_COOKIE_SALT	signed encrypted cookie
ACTION_DISPATCH_HTTP_AUTH_SALT	http authentication
ACTION_DISPATCH_KEY_GENERATOR	#<ActiveSupport::CachingKeyGenerator:0x00561f3c0baf48>
ACTION_DISPATCH_LOGGER	#<ActiveSupport::Logger:0x00561f38c24dd8>
ACTION_DISPATCH_PARAMETER_FILTER	["password"]
ACTION_DISPATCH_REDIRECT_FILTER	[]
ACTION_DISPATCH_REMOTE_IP	8.8.8.8
ACTION_DISPATCH_REQUEST_ACCEPTS	["*/*"]
ACTION_DISPATCH_REQUEST_CONTENT_TYPE	text/plain
ACTION_DISPATCH_REQUEST_FORMATS	["*/*"]
ACTION_DISPATCH_REQUEST_ID	bfc15d3c-684e-4fff-a041-33086d735294
ACTION_DISPATCH_REQUEST_PARAMETERS	{"key"=>"***key***", "controller"=>"api/v3/notices", "action"=>"create", "project_id"=>"2"}
ACTION_DISPATCH_REQUEST_PATH_PARAMETERS	{"controller"=>"api/v3/notices", "action"=>"create", "project_id"=>"2"}
ACTION_DISPATCH_REQUEST_QUERY_PARAMETERS	{"key"=>"***key***"}
ACTION_DISPATCH_REQUEST_REQUEST_PARAMETERS	
ACTION_DISPATCH_REQUEST_UNSIGNED_SESSION_COOKIE	
ACTION_DISPATCH_ROUTES	#<ActionDispatch::Routing::RouteSet:0x00561f38aac438>
ACTION_DISPATCH_SECRET_KEY_BASE	f...c
ACTION_DISPATCH_SECRET_TOKEN	
ACTION_DISPATCH_SHOW_DETAILED_EXCEPTIONS	false
ACTION_DISPATCH_SHOW_EXCEPTIONS	true
ACTION_DISPATCH_SIGNED_COOKIE_SALT	signed cookie
PUMA_CONFIG	#<Puma::Configuration:0x00561f3c6695e8>
PUMA_SOCKET	#<TCPSocket:0x00561f37929788>
RACK_AFTER_REPLY	[]
RACK_ERRORS	#<IO:0x00561f3570dc08>
RACK_HIJACK	#<Puma::Client:0x00561f37929670>
RACK_HIJACK?	true
RACK_INPUT	#<StringIO:0x00561f37b8cca0>
RACK_MULTIPROCESS	false
RACK_MULTITHREAD	true
RACK_REQUEST_COOKIE_HASH	
RACK_REQUEST_QUERY_HASH	{"key"=>"***key***"}
RACK_REQUEST_QUERY_STRING	key=***key***
RACK_RUN_ONCE	false
RACK_SESSION	
RACK_SESSION_OPTIONS	{"path"=>"/", "domain"=>nil, "expire_after"=>nil, "secure"=>"false", "httponly"=>"true", "defer"=>"false", "renew"=>"false", "id"=>nil}
RACK_URL_SCHEME	http
RACK_VERSION	["1", "3"]
WARDEN	Warden::Proxy:47346044698260 @config={:default_scope=>:user, :scope_defaults=>{}, :default_strategies=>{:user=>[:rememberable, :database_authenticatable]}, :intercept_401=>false, :failure_app=>#<Devise::Delegator:0x00561f38dc68d0>}

```

**Comment by rud** [comment]
Expecting this to also be fixed by #1311
